crates/stackable-operator/src/kvp/mod.rs

```diff
@@ -90,265 +94,102 @@ where
 ///
 /// - <https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/>
 /// - <https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/>
-#[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord)]
+#[derive(Clone, PartialEq, Eq, PartialOrd, Ord)]
 pub struct KeyValuePair<T>
 where
     T: Value,
 {
-    key: Key,
-    value: T,
+    pub key: Key,
+    pub value: T,
 }
 
-impl<K, V, T> TryFrom<(K, V)> for KeyValuePair<T>
+impl<V> TryFrom<(&str, &str)> for KeyValuePair<V>
 where
-    K: AsRef<str>,
-    V: AsRef<str>,
-    T: Value,
+    V: Value,
 {
-    type Error = KeyValuePairError<T::Error>;
-
-    fn try_from(value: (K, V)) -> Result<Self, Self::Error> {
-        let key = Key::from_str(value.0.as_ref()).context(InvalidKeySnafu {
-            key: value.0.as_ref(),
-        })?;
-
-        let value = T::from_str(value.1.as_ref()).context(InvalidValueSnafu {
-            key: key.to_string(),
-            value: value.1.as_ref(),
-        })?;
+    type Error = KeyValuePairError<V::Error>;
 
+    fn try_from((key, value): (&str, &str)) -> Result<Self, Self::Error> {
+        let key = Key::from_str(key).context(InvalidKeySnafu { key })?;
+        let value = V::from_str(value).context(InvalidValueSnafu { key: &key, value })?;
         Ok(Self { key, value })
     }
 }
 
-impl<T> Display for KeyValuePair<T>
-where
-    T: Value,
-{
-    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        write!(f, "{}={}", self.key, self.value)
+impl<V: Value> From<KeyValuePair<V>> for (Key, V) {
+    fn from(KeyValuePair { key, value }: KeyValuePair<V>) -> Self {
+        (key, value)
     }
 }
 
-impl<T> KeyValuePair<T>
-where
-    T: Value,
-{
-    /// Creates a new [`KeyValuePair`] from a validated [`Key`] and value.
-    pub fn new(key: Key, value: T) -> Self {
-        Self { key, value }
-    }
-
-    /// Returns an immutable reference to the pair's [`Key`].
-    pub fn key(&self) -> &Key {
-        &self.key
-    }
-
-    /// Returns an immutable reference to the pair's value.
-    pub fn value(&self) -> &T {
-        &self.value
+impl<T: Value> Display for KeyValuePair<T> {
```

Comment:
nit: Use `V` instead of `T` to be consistent.

```suggestion
impl<V: Value> Display for KeyValuePair<V> {
```

Reply:
17458e1afd86e674a63cd99c3fabcc3261786192, must've missed updating that...